plugins/azure-devops/src/api/AzureDevOpsClient.ts

```diff
@@ -46,6 +48,17 @@ export class AzureDevOpsClient implements AzureDevOpsApi {
     return { items };
   }
 
+  public async getPullRequests(
+    projectName: string,
+    repoName: string,
+    options?: PullRequestOptions,
+  ): Promise<{ items: PullRequest[] }> {
+    const items = await this.get<PullRequest[]>(
+      `pull-requests/${projectName}/${repoName}?top=${options?.top}&status=${options?.status}`,
```

Comment:
Whenever forming a URL, you should do `encodeURIComponent` on the string so that a malicious user can't craft URLs that break the patterns and worst case cause trouble.

I see that the `top` and `status` fieds are optional, but still are added unconditionally to the URL which probably means that the string "undefined" will be in the URL. Might I suggest doing something like this?

```ts
const q = new URLSearchParams();
if (options?.top) {
  q.append('top', options.top);
}
if (options?.status) {
  q.append('status', options.status;
}
const url = `pull-requests/${encodeURIComponent(projectName)}/${encodeURIComponent(repoName)}?${q}`
```

(both of these apply to the other method above too of course).

Reply:
Yes, this is a big gap in my knowledge so thanks for pointing this out. Both have been updated with this and I'll know to do this in the future as well 👍

Reply:
Nice! Thanks